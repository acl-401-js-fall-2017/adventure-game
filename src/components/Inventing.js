import React, { Component } from 'react';
import inventPic from '../pics/testing.png';
import PropTypes from 'prop-types';


export default class Inventing extends Component {
  constructor() {
    super();
    this.state = {
      kids: [],
      gotGob: false
    };
  }

  handleClick(value, inv){
    const { handleInventory, inventory } = this.props;
    const kids = [...this.state.kids];
    if(this.state.kids.length <= 3 && inv === 'kids') {
      kids.push('kids');
    }
    const gotGob = inventory.indexOf('gobstopper') === -1 && inv === 'gobstopper' && kids.length >= 4 ? true : false;
    this.setState({
      kids,
      gotGob
    },() =>  {
      if (inventory.indexOf('gobstopper') === -1 && inv === 'gobstopper' && this.state.kids.length >= 4) {
        handleInventory('gobstopper');
      }
    });
  }


  render() {
    let message = 'Give Mr Wonka the loompas for the gobstopper';
    if(this.state.length < 4) {
      message = 'There are still children in your way';
    } else if(this.state.gotGob === true) {
      message = 'You got the gobstopper!';
    }
    
    return (
      <section className="challenge">
        <img src={inventPic} alt="" useMap="#invMap" />
        <map name="invMap" id="Map">
          <area alt="grid" data-value="correct" data-inv="gobstopper" shape="poly" coords="436,249,436,295,486,295,486,242" onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          <area alt="grid" data-value="person" data-inv="kids" shape="poly" coords="6,432,8,635,163,631,222,442" onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          <area alt="grid" data-value="person" data-inv="kids" shape="poly" coords="246,324,207,613,458,618,383,321" onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          <area alt="grid" data-value="person" data-inv="kids" shape="poly" coords="467,609,580,304,743,324,784,608" onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
          <area alt="grid" data-value="person" data-inv="kids" shape="poly" coords="795,602,918,295,1085,292,1142,590"   onClick={({ target }) => this.handleClick(target.dataset.value, target.dataset.inv)}/>
        </map>
        {this.state.kids.length < 4 && (
          <p className="mission"> 
          Move the { 4 - this.state.kids.length} kids out of the way to get the everlasting gobstopper.
          </p>
        )}
        <p className="message">
          Message: {message} 
        </p>
      </section>
    );
  }

}

Inventing.PropTypes = {
  message: PropTypes.string
};
