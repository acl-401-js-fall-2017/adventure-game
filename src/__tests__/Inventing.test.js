import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Inventing from '../components/Inventing';

describe('inventing', () => {

  it('shows a Inventing', () => {
    const wrapper = shallow(<Inventing handleInventory={() => {}} inventory={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});