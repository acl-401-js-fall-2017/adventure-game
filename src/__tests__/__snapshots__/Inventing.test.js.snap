// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inventing shows a Inventing 1`] = `
<section
  className="challenge"
>
  <img
    alt=""
    src="testing.png"
    useMap="#invMap"
  />
  <map
    id="Map"
    name="invMap"
  >
    <area
      alt="grid"
      coords="436,249,436,295,486,295,486,242"
      data-inv="gobstopper"
      data-value="correct"
      onClick={[Function]}
      shape="poly"
    />
    <area
      alt="grid"
      coords="6,432,8,635,163,631,222,442"
      data-inv="kids"
      data-value="person"
      onClick={[Function]}
      shape="poly"
    />
    <area
      alt="grid"
      coords="246,324,207,613,458,618,383,321"
      data-inv="kids"
      data-value="person"
      onClick={[Function]}
      shape="poly"
    />
    <area
      alt="grid"
      coords="467,609,580,304,743,324,784,608"
      data-inv="kids"
      data-value="person"
      onClick={[Function]}
      shape="poly"
    />
    <area
      alt="grid"
      coords="795,602,918,295,1085,292,1142,590"
      data-inv="kids"
      data-value="person"
      onClick={[Function]}
      shape="poly"
    />
  </map>
  <p
    className="mission"
  >
    Move the 
    4
     kids out of the way to get the everlasting gobstopper.
  </p>
  <p
    className="message"
  >
    Message: 
    Give Mr Wonka the loompas for the gobstopper
  </p>
</section>
`;
